client/js/src/client.ts

```diff
@@ -433,6 +434,54 @@ export class Client {
 			component_server: this.component_server
 		};
 	}
+
+	private async connect_ws(url: string): Promise<void> {
+		return new Promise((resolve, reject) => {
+			const ws = new WebSocket(url);
```

Comment:
I would try catch this b/c if websocket support is not available in the browser/environment, this constructor might raise an exception, in that case, we can set this.ws_map[url] = "failed";